File: Quiz.Task/src/Quiz.Task.Service/Models/Folder.cs
using Quiz.Common.Models;

namespace Quiz.Task.Service.Models
{
    public class Folder : IModel
    {
        public int Id { get; set; }
        public string FolderName { get; set; } 
        public DateTime CreateDate { get; set; }
        public int UserId { get; set; }
        public int ClassId { get; set; }
        public virtual UserInfo User { get; set; }
        public virtual Class Class { get; set; }
        public virtual ICollection<FolderDetail> FolderDetails { get; set; }
    
    }    
}

File: quiz/lib/models/folder.dart
import 'package:intl/intl.dart';
import 'package:quiz/models/class.dart';
import 'package:quiz/models/user.dart';

class Folder{
  int folderId;
  String folderName;
  DateTime? createDate;
  UserInfo user;
  Class? e_class;

  Folder({required this.folderId, required this.folderName, required this.user, this.createDate, this.e_class});
  factory Folder.fromJson(Map<String, dynamic> json) => Folder(
    folderId: json['id'],
    folderName: json['folderName'],
    createDate: json['createDate']!= null? DateFormat('dd-MM-yyyy').parse(json['createDate']): null,
    user: UserInfo.fromJson(json['user']),
    e_class:  json['e_class']!=null? Class.fromJson(json['class']): null
  );
}

class FolderRequest{
  int folderId;
  String folderName;
  int userId;
  int classId;

  FolderRequest({required this.folderId, required this.folderName, required this.userId, required this.classId});
}

